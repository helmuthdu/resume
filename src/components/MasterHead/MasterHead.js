// @flow
import React from 'react';
import { Button, Container, Divider, Header, Segment } from 'semantic-ui-react';
import { menuItems } from '../../layouts/DefaultLayout';
import HeaderMenu from '../HeaderMenu/HeaderMenu';

export const MasterHead = ({ pathname }: any) => (
  <Segment vertical inverted textAlign="center" className="masthead">
    <HeaderMenu pathname={pathname} items={menuItems} pointing secondary inverted visible />
    <Container text>
      <Header inverted as="h1">
        Helmuth Saatkamp
      </Header>
      <Divider hidden />
      <Header inverted as="h2">
        Web Developer
      </Header>
      <Button
        primary
        inverted
        size="large"
        as="a"
        href={`${process.env.PUBLIC_URL || ''}/docs/helmuth_saatkamp_resume_en.pdf`}>
        Download My CV
      </Button>
    </Container>
  </Segment>
);

export default MasterHead;
