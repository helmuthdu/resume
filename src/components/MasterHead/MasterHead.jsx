import PropTypes from 'prop-types';
import React from 'react';
import { Button, Container, Divider, Header, Segment } from 'semantic-ui-react';
import { menuItems } from '../../layouts/DefaultLayout';
import HeaderMenu from '../HeaderMenu/HeaderMenu';

export const MasterHead = ({ pathname }) => (
  <Segment vertical inverted textAlign="center" className="masthead">
    <HeaderMenu pathname={pathname} items={menuItems} pointing secondary inverted visible />
    <Container text>
      <Header inverted as="h1">
        Helmuth Saatkamp
      </Header>
      <Divider hidden />
      <Header inverted as="h2">
        Software Engineer
      </Header>
      <Button
        color={'blue'}
        inverted
        size="large"
        as="a"
        href="https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf">
        Download My CV
      </Button>
    </Container>
  </Segment>
);

MasterHead.propTypes = {
  pathname: PropTypes.string.isRequired
};

export default MasterHead;
