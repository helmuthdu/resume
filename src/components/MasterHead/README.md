# MasterHead component

MasterHead Section.

## Source

    <MasterHead id={name}/>
