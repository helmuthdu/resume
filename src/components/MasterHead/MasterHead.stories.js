const withReadme = require('storybook-readme/with-readme').default;
const AboutReadme = require('./README.md');

import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import MasterHead from './MasterHead';

storiesOf('MasterHead', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(AboutReadme))
  .add('default', () => {
    return <MasterHead />;
  });
