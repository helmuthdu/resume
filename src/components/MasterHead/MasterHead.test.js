import { shallow } from 'enzyme';
import 'jest';
import React from 'react';
import { MasterHead } from './MasterHead';

describe('MasterHead component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<MasterHead pathname="/" />);
    expect(wrapper).toMatchSnapshot();
  });
});
