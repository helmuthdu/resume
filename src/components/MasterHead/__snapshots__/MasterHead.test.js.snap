// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component -> MasterHead should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MasterHead
    pathname="/"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(HeaderMenu)
          inverted={true}
          items={
            Array [
              Object {
                "exact": true,
                "icon": "home",
                "inverted": true,
                "name": "Home",
                "path": "/",
              },
              Object {
                "exact": false,
                "icon": "info circle",
                "name": "About",
                "path": "#about",
              },
              Object {
                "exact": false,
                "icon": "newspaper",
                "name": "Resume",
                "path": "#resume",
              },
              Object {
                "exact": false,
                "icon": "newspaper",
                "name": "Portfolio",
                "path": "#portfolio",
              },
              Object {
                "exact": false,
                "icon": "newspaper",
                "name": "Contact",
                "path": "#contact",
              },
            ]
          }
          pathname="/"
          pointing={true}
          secondary={true}
          visible={true}
        />,
        <Container
          text={true}
        >
          <Header
            as="h1"
            inverted={true}
          >
            Helmuth Saatkamp
          </Header>
          <Divider
            hidden={true}
          />
          <Header
            as="h2"
            inverted={true}
          >
            Web Developer
          </Header>
          <Button
            as="a"
            href="https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf"
            inverted={true}
            primary={true}
            role="button"
            size="large"
          >
            Download My CV
          </Button>
        </Container>,
      ],
      "className": "masthead",
      "inverted": true,
      "textAlign": "center",
      "vertical": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inverted": true,
          "items": Array [
            Object {
              "exact": true,
              "icon": "home",
              "inverted": true,
              "name": "Home",
              "path": "/",
            },
            Object {
              "exact": false,
              "icon": "info circle",
              "name": "About",
              "path": "#about",
            },
            Object {
              "exact": false,
              "icon": "newspaper",
              "name": "Resume",
              "path": "#resume",
            },
            Object {
              "exact": false,
              "icon": "newspaper",
              "name": "Portfolio",
              "path": "#portfolio",
            },
            Object {
              "exact": false,
              "icon": "newspaper",
              "name": "Contact",
              "path": "#contact",
            },
          ],
          "pathname": "/",
          "pointing": true,
          "secondary": true,
          "visible": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Header
              as="h1"
              inverted={true}
            >
              Helmuth Saatkamp
            </Header>,
            <Divider
              hidden={true}
            />,
            <Header
              as="h2"
              inverted={true}
            >
              Web Developer
            </Header>,
            <Button
              as="a"
              href="https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf"
              inverted={true}
              primary={true}
              role="button"
              size="large"
            >
              Download My CV
            </Button>,
          ],
          "text": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h1",
              "children": "Helmuth Saatkamp",
              "inverted": true,
            },
            "ref": null,
            "rendered": "Helmuth Saatkamp",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "hidden": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h2",
              "children": "Web Developer",
              "inverted": true,
            },
            "ref": null,
            "rendered": "Web Developer",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "a",
              "children": "Download My CV",
              "href": "https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf",
              "inverted": true,
              "primary": true,
              "role": "button",
              "size": "large",
            },
            "ref": null,
            "rendered": "Download My CV",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(HeaderMenu)
            inverted={true}
            items={
              Array [
                Object {
                  "exact": true,
                  "icon": "home",
                  "inverted": true,
                  "name": "Home",
                  "path": "/",
                },
                Object {
                  "exact": false,
                  "icon": "info circle",
                  "name": "About",
                  "path": "#about",
                },
                Object {
                  "exact": false,
                  "icon": "newspaper",
                  "name": "Resume",
                  "path": "#resume",
                },
                Object {
                  "exact": false,
                  "icon": "newspaper",
                  "name": "Portfolio",
                  "path": "#portfolio",
                },
                Object {
                  "exact": false,
                  "icon": "newspaper",
                  "name": "Contact",
                  "path": "#contact",
                },
              ]
            }
            pathname="/"
            pointing={true}
            secondary={true}
            visible={true}
          />,
          <Container
            text={true}
          >
            <Header
              as="h1"
              inverted={true}
            >
              Helmuth Saatkamp
            </Header>
            <Divider
              hidden={true}
            />
            <Header
              as="h2"
              inverted={true}
            >
              Web Developer
            </Header>
            <Button
              as="a"
              href="https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf"
              inverted={true}
              primary={true}
              role="button"
              size="large"
            >
              Download My CV
            </Button>
          </Container>,
        ],
        "className": "masthead",
        "inverted": true,
        "textAlign": "center",
        "vertical": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inverted": true,
            "items": Array [
              Object {
                "exact": true,
                "icon": "home",
                "inverted": true,
                "name": "Home",
                "path": "/",
              },
              Object {
                "exact": false,
                "icon": "info circle",
                "name": "About",
                "path": "#about",
              },
              Object {
                "exact": false,
                "icon": "newspaper",
                "name": "Resume",
                "path": "#resume",
              },
              Object {
                "exact": false,
                "icon": "newspaper",
                "name": "Portfolio",
                "path": "#portfolio",
              },
              Object {
                "exact": false,
                "icon": "newspaper",
                "name": "Contact",
                "path": "#contact",
              },
            ],
            "pathname": "/",
            "pointing": true,
            "secondary": true,
            "visible": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Header
                as="h1"
                inverted={true}
              >
                Helmuth Saatkamp
              </Header>,
              <Divider
                hidden={true}
              />,
              <Header
                as="h2"
                inverted={true}
              >
                Web Developer
              </Header>,
              <Button
                as="a"
                href="https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf"
                inverted={true}
                primary={true}
                role="button"
                size="large"
              >
                Download My CV
              </Button>,
            ],
            "text": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h1",
                "children": "Helmuth Saatkamp",
                "inverted": true,
              },
              "ref": null,
              "rendered": "Helmuth Saatkamp",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "hidden": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h2",
                "children": "Web Developer",
                "inverted": true,
              },
              "ref": null,
              "rendered": "Web Developer",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "a",
                "children": "Download My CV",
                "href": "https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf",
                "inverted": true,
                "primary": true,
                "role": "button",
                "size": "large",
              },
              "ref": null,
              "rendered": "Download My CV",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
